kitty/fonts.c

```diff
@@ -741,7 +741,16 @@ shape(CPUCell *first_cpu_cell, GPUCell *first_gpu_cell, index_type num_cells, hb
     group_state.last_cpu_cell = first_cpu_cell + (num_cells ? num_cells - 1 : 0);
     group_state.last_gpu_cell = first_gpu_cell + (num_cells ? num_cells - 1 : 0);
     load_hb_buffer(first_cpu_cell, first_gpu_cell, num_cells);
-    hb_shape(font, harfbuzz_buffer, NULL, 0);
+
+    if (!render_ligature) {
+        hb_feature_t feature;
+        const char* feature_str = "-calt";
+        if (!hb_feature_from_string(feature_str, strlen(feature_str), &feature)) fatal("hb_feature_from_string() failed");
```

Comment:
These seems unnecessarily ineeficient. Does it really need to be called on every shape() run. Cant it instead use  a static variable? And there should be no need to use strlen() here, sizeof() should suffice.

Reply:
Where is the best place to initialise that variable? `sizeof()` won't work here, as it will return the size of the pointer (8 bytes on my machine) instead of the length of the actual string.

Reply:
Maybe I found a good place.

Reply:
you know the relation between sizeof and strlen() it is fixed, so there is not need to use strlen.